src/git.ts

```diff
@@ -142,17 +130,14 @@ export interface GitEvents {
    * no listeners, `head.accept()` is called automatically.
    *
    **/
-  on(event: 'head', listener: (head: HeadData) => void): this;
+  on(event: 'head', listener: (head: HeadData<T>) => void): this;
 }
-export class Git extends EventEmitter implements GitEvents {
+export class Git<T = any> extends EventEmitter implements GitEvents<T> {
```

Comment:
Can we add a comment on why this is any?

Reply:
Yeah, I'm not decided on this totally. I guess it could should probably default to undefined/void

The TS will warn you to add the type to your Git constructor if you return something from `authenticate`